pkg/controller/pod_volume_restore_controller_test.go

```diff
@@ -0,0 +1,572 @@
+	assert.False(t, isPodOnNode(pod, "bar"))
+
+	pod.Spec.NodeName = "foo"
+	assert.False(t, isPodOnNode(pod, "bar"))
+
+	pod.Spec.NodeName = "bar"
+	assert.True(t, isPodOnNode(pod, "bar"))
+}
+
+func TestIsResticContainerRunning(t *testing.T) {
+	tests := []struct {
+		name     string
+		pod      *corev1api.Pod
+		expected bool
+	}{
+		{
+			name: "pod with no init containers should return false",
+			pod: &corev1api.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "ns-1",
+					Name:      "pod-1",
+				},
+			},
+			expected: false,
+		},
+		{
+			name: "pod with running init container that's not restic should return false",
+			pod: &corev1api.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "ns-1",
+					Name:      "pod-1",
+				},
+				Spec: corev1api.PodSpec{
+					InitContainers: []corev1api.Container{
+						{
+							Name: "non-restic-init",
+						},
+					},
+				},
+				Status: corev1api.PodStatus{
+					InitContainerStatuses: []corev1api.ContainerStatus{
+						{
+							State: corev1api.ContainerState{
+								Running: &corev1api.ContainerStateRunning{StartedAt: metav1.Time{Time: time.Now()}},
+							},
+						},
+					},
+				},
+			},
+			expected: false,
+		},
+		{
+			name: "pod with running restic init container that's not first should return false",
+			pod: &corev1api.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "ns-1",
+					Name:      "pod-1",
+				},
+				Spec: corev1api.PodSpec{
+					InitContainers: []corev1api.Container{
+						{
+							Name: "non-restic-init",
+						},
+						{
+							Name: restic.InitContainer,
+						},
+					},
+				},
+				Status: corev1api.PodStatus{
+					InitContainerStatuses: []corev1api.ContainerStatus{
+						{
+							State: corev1api.ContainerState{
+								Running: &corev1api.ContainerStateRunning{StartedAt: metav1.Time{Time: time.Now()}},
+							},
+						},
+						{
+							State: corev1api.ContainerState{
+								Running: &corev1api.ContainerStateRunning{StartedAt: metav1.Time{Time: time.Now()}},
+							},
+						},
+					},
+				},
+			},
+			expected: false,
+		},
+		{
+			name: "pod with running restic init container as first initContainer should return true",
+			pod: &corev1api.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "ns-1",
+					Name:      "pod-1",
+				},
+				Spec: corev1api.PodSpec{
+					InitContainers: []corev1api.Container{
+						{
+							Name: restic.InitContainer,
+						},
+						{
+							Name: "non-restic-init",
+						},
+					},
+				},
+				Status: corev1api.PodStatus{
+					InitContainerStatuses: []corev1api.ContainerStatus{
+						{
+							State: corev1api.ContainerState{
+								Running: &corev1api.ContainerStateRunning{StartedAt: metav1.Time{Time: time.Now()}},
+							},
+						},
+						{
+							State: corev1api.ContainerState{
+								Running: &corev1api.ContainerStateRunning{StartedAt: metav1.Time{Time: time.Now()}},
+							},
+						},
+					},
+				},
+			},
+			expected: true,
+		},
```

Comment:
Need a test case for first initContainer is our restic, but container state != running?